python/tank/platform/engine.py

```diff
@@ -206,15 +236,78 @@ def get_env(self):
 
     ##########################################################################################
     # properties used by internal classes, not part of the public interface
-    
+
+    def __create_log_handler(self):
+        """
+        Creates a std python logging LogHandler
+        that dispatches all log messages to the
+        :meth:`Engine._emit_log_message()` method in a threadsafe manner.
+
+        :return: :class:`python.logging.LogHandler` object
+        """
+        # use a thin custom wrapper that dispatches to the callback
+
+        # alias 'self' so we can refer to the engine
+        # object from inside our wrapper class
+        engine_instance = self
+
+        # set up logging handler class
```

Comment:
it was two lines of code at first... then it inevitably grew... now it's in it's own file alongside another handler which offers legacy support.